models/articles-model.js

```diff
@@ -17,3 +17,24 @@ exports.fetchArticlesbyId = (id) => {
       return rows;
     });
 };
+exports.fetchArticles = () => {
+  return db
+    .query(
+      `SELECT author,title,article_id,topic,created_at,votes,article_img_url FROM articles
+      ORDER BY created_at DESC`
+    )
+    .then(({ rows }) => {
+      const articles = rows;
+      const query = `SELECT * FROM comments`;
+      return Promise.all([articles, db.query(query)]);
+    })
+    .then(([articles, response]) => {
+      const comments = response.rows;
+      return articles.map((article) => {
+        article.comment_count = comments.filter((comment) => {
+          return comment.article_id === article.article_id;
+        }).length;
+        return article;
+      });
```

Comment:
You can use a single SQL query to join the articles and comments tables instead of running two separate queries. 💡